tools/test_determinism.py

```
@@ -0,0 +1,98 @@
+    std_b = res_b.groupby(config.COL_GROUP)["Score2"].mean().std(ddof=1)
+
+    print(f"W1=1, W2=0: Status={m_a['status']}, Std1={std_a:.6f}")
+    print(f"W1=0, W2=1: Status={m_b['status']}, Std2={std_b:.6f}")
+
+    if res_a[config.COL_GROUP].tolist() == res_b[config.COL_GROUP].tolist():
+        print("✅ SUCCESS: Weight toggle is deterministic.")
+    else:
+        print("❌ FAILURE: Weight toggle produced different results!")
+
+    print("\n>>> DETERMINISM TEST: WARM START CONTINUITY")
+    res_c, m_c = OptimizationService.run(
```

Ruff (0.15.12)

[warning] Unpacked variable `m_c` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)